azure/README.md

```diff
@@ -330,25 +307,66 @@ kubectl label nodes --all node.kubernetes.io/worker=
 > [configure](https://projectcalico.docs.tigera.io/networking/vxlan-ipip#configure-vxlan-encapsulation-for-all-inter-workload-traffic)
 > VXLAN encapsulation for all inter workload traffic.
 
+### User Assigned Identity and Federated Credentials
+
+We will use a Workload Identity to provide the CAA DaemonSet with the
+privileges it requires to manage virtual machines in our resource group.
+Note: if you use an existing AKS cluster it might need to be configured
+to support Workload Identity and OIDC, please refer to the instructions
+in [this guide](https://learn.microsoft.com/en-us/azure/aks/workload-identity-deploy-cluster#update-an-existing-aks-cluster).
+
+```bash
+az identity create --name caa-identity -g "$AZURE_RESOURCE_GROUP" -l "$AZURE_REGION"
+export USER_ASSIGNED_CLIENT_ID="$(az identity show --resource-group "$AZURE_RESOURCE_GROUP" --name caa-identity --query 'clientId' -otsv)"
+```
+
+We'll annotate the CAA Service Account with the Workload Identity's
+`CLIENT_ID` and make the CAA DaemonSet use Workload Identity
+for authentication:
+
+```bash
+cat <<EOF > install/overlays/azure/workload-identity.yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: cloud-api-adaptor-daemonset
+  namespace: confidential-containers-system
+spec:
+  template:
+    metadata:
+      labels:
+        azure.workload.identity/use: "true"
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cloud-api-adaptor
+  namespace: confidential-containers-system
+  annotations:
+    azure.workload.identity/client-id: "$USER_ASSIGNED_CLIENT_ID"
+```
+
 ### AKS Resource Group permissions
 
-AKS deploys the actual resources like the worker nodes in another resource
-group named in environment variable `AKS_RG`. For the CAA to be able to create
-pod VM in the same subnet as the worker nodes of the AKS cluster, run the
-following command:
+For CAA to be able to manage VMs we need to assign the identity two contributor
+roles, scoped to the Resource Group:
 
 ```bash
-az ad sp create-for-rbac \
-  -n "caa-${AZURE_RESOURCE_GROUP}" \
-  --role Contributor \
-  --scopes "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${AKS_RG}" \
-  --query "password"
+az role assignment create --role "Virtual Machine Contributor" --assignee "$USER_ASSIGNED_CLIENT_ID" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourcegroups/${AZURE_RESOURCE_GROUP}"
+az role assignment create --role "Network Contributor" --assignee "$USER_ASSIGNED_CLIENT_ID" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourcegroups/${AZURE_RESOURCE_GROUP}"
```

Comment:
```suggestion
az role assignment create \
    --role "Contributor" \
    --assignee "$USER_ASSIGNED_CLIENT_ID" \
    --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourcegroups/${AZURE_RESOURCE_GROUP}"

az role assignment create \
    --role "Contributor" \
    --assignee "$USER_ASSIGNED_CLIENT_ID" \
    --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}/resourcegroups/${AKS_RG}"
```

Reply:
We don't need a full Contributor Role, do we?

Reply:
I just verified it with a new cluster, we only need network contributor on the aks rg.